browser/ui/webui/settings/brave_search_engines_handler.cc

```diff
@@ -28,9 +31,17 @@ constexpr char kBraveSearchForTorKeyword[] =
     ":search.brave4u7jddbv7cyviptqjc7jusxh72uik7zt6adtckl5f4nwy2v72qd.onion";
 
 // Put yahoo at first place.
-void SortDefaultSearchEnginesListInJP(base::Value::List& defaults) {
-  auto* local_state = g_browser_process->local_state();
-  if (!local_state || brave_l10n::GetCountryCode(local_state) != "JP") {
+void SortDefaultSearchEnginesListInJP(base::Value::List& defaults,
+                                      Profile* profile) {
+  // Only apply this logic for JP locale
+  auto* regional_capabilities =
+      regional_capabilities::RegionalCapabilitiesServiceFactory::GetForProfile(
```

Comment:
please use dependency injection, RegionalCapabilitiesService should be passed in to the constructor for BraveSearchEnginesHandler

Reply:
very strange to me that this is an anonymous method as well 🤷

Reply:
Fixed